### DISCUSSION TITLE
tmp folder file gone

### DISCUSSION THREAD

**Discussion by autumn31** [vercel/community#2071] [Help] [answered]
### Problem Description

I've create file in /tmp folder for future access, however it's will be gone for a while. Wonder what is the expiration mechanism for the file under /tmp folder, is it auto cleanup via each new deployment or for a range of time

**Comment by amyegan** [comment, marked as answer]
Hi @autumn31! The `/tmp` folder does not persist. It is only intended for temporary use, and a long-term storage storage solution should be used for things you intend to keep. You can find an example of one of the suggested uses for the `/tmp` folder in our guide on [using temporary storage](https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-temporary-storage).

Some other related guides are available as well:
- [How can I use AWS S3 with Vercel?](https://vercel.com/guides/how-can-i-use-aws-s3-with-vercel)
- [Using Databases with Vercel](https://vercel.com/guides/using-databases-with-vercel)
- [How to Upload and Store Files with Vercel](https://vercel.com/guides/how-to-upload-and-store-files-with-vercel)
